### PR TITLE
Add SQL blob auditing policies

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

**Comment by irmatov** [comment]
WIP/ to be reviewed after previous PRs.

**Review by fdistorted — APPROVED** [review]
LGTM

File: resources/sql_databases.go [hunk 1]
```diff
@@ -316,6 +316,97 @@ func SQLDatabases() *schema.Table {
...
...ould not be used in combination with other groups as this will result in duplicate audit logs  For more information, see [Database-Level Audit Action Groups](https://docsmicrosoftcom/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups)  For Database auditing policy, specific Actions can also be specified (note that Actions cannot be specified for Server auditing policy) The supported actions to audit are: SELECT UPDATE INSERT DELETE EXECUTE RECEIVE REFERENCES  The general form for defining an action to be audited is: {action} ON {object} BY {principal}  Note that <object> in the above format can refer to an object like a table, view, or stored procedure, or an entire database or schema For the latter cases, the forms DATABASE::{db_name} and SCHEMA::{schema_name} are used, respectively  For example: SELECT on dbomyTable by public SELECT on DATABASE::myDatabase by public SELECT on SCHEMA::mySchema by public  For more information, see [Database-Level Audit Actions](https://docsmicrosoftcom/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)",
```
Comment (roneli) [inline resources/sql_databases.go:363]: shorten this description

File: resources/sql_servers.go [hunk 2]
```diff
@@ -280,6 +280,90 @@ func SQLServers() *schema.Table {
...
+						Type:        schema.TypeBool,
+						Resolver:    schema.PathResolver("ServerBlobAuditingPolicyProperties.IsStorageSecondaryKeyInUse"),
+					},
+					{
+						Name:        "is_azure_monitor_target_enabled",
+						Description: "Specifies whether audit events are sent to Azure Monitor In order to send the events to Azure Monitor, specify 'state' as 'Enabled' and 'isAzureMonitorTargetEnabled' as true  When using REST API to configure auditing, Diagnostic Settings with 'SQLSecurityAuditEvents' diagnostic logs category on the database should be also created Note that for server level audit you should use the 'master' database as {databaseName}  Diagnostic Settings URI format: PUT https://managementazurecom/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/MicrosoftSql/servers/{serverName}/databases/{databaseName}/providers/microsoftinsights/diagnosticSettings/{settingsName}?api-version=2017-05-01-preview  For more information, see [Diagnostic Settings REST API](https://gomicrosoftcom/fwlink/?linkid=2033207) or [Diagnostic Settings PowerShell](https://gomicrosoftcom/fwlink/?linkid=2033043)",
```
Comment (roneli) [inline resources/sql_servers.go:339]: shorten description

File: resources/sql_servers.go [hunk 3]
```diff
@@ -280,6 +280,90 @@ func SQLServers() *schema.Table {
...
...ould not be used in combination with other groups as this will result in duplicate audit logs  For more information, see [Database-Level Audit Action Groups](https://docsmicrosoftcom/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups)  For Database auditing policy, specific Actions can also be specified (note that Actions cannot be specified for Server auditing policy) The supported actions to audit are: SELECT UPDATE INSERT DELETE EXECUTE RECEIVE REFERENCES  The general form for defining an action to be audited is: {action} ON {object} BY {principal}  Note that <object> in the above format can refer to an object like a table, view, or stored procedure, or an entire database or schema For the latter cases, the forms DATABASE::{db_name} and SCHEMA::{schema_name} are used, respectively  For example: SELECT on dbomyTable by public SELECT on DATABASE::myDatabase by public SELECT on SCHEMA::mySchema by public  For more information, see [Database-Level Audit Actions](https://docsmicrosoftcom/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)",
```
Comment (roneli) [inline resources/sql_servers.go:321]: shorten description

File: resources/sql_servers.go [hunk 4]
```diff
@@ -280,6 +280,90 @@ func SQLServers() *schema.Table {
...
+					},
+					{
+						Name:        "storage_endpoint",
+						Description: "Specifies the blob storage endpoint (eg https://MyAccountblobcorewindowsnet) If state is Enabled, storageEndpoint or isAzureMonitorTargetEnabled is required",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("ServerBlobAuditingPolicyProperties.StorageEndpoint"),
+					},
+					{
+						Name:        "storage_account_access_key",
+						Description: "Specifies the identifier key of the auditing storage account If state is Enabled and storageEndpoint is specified, not specifying the storageAccountAccessKey will use SQL server system-assigned managed identity to access the storage Prerequisites for using managed identity authentication: 1 Assign SQL Server a system-assigned managed identity in Azure Active Directory (AAD) 2 Grant SQL Server identity access to the storage account by adding 'Storage Blob Data Contributor' RBAC role to the server identity For more information, see [Auditing to storage using Managed Identity authentication](https://gomicrosoftcom/fwlink/?linkid=2114355)",
```
Comment (roneli) [inline resources/sql_servers.go:309]: shorten description

File: resources/sql_databases.go [hunk 5]
```diff
@@ -316,6 +316,97 @@ func SQLDatabases() *schema.Table {
...
+						Type:        schema.TypeBool,
+						Resolver:    schema.PathResolver("DatabaseBlobAuditingPolicyProperties.IsStorageSecondaryKeyInUse"),
+					},
+					{
+						Name:        "is_azure_monitor_target_enabled",
+						Description: "Specifies whether audit events are sent to Azure Monitor In order to send the events to Azure Monitor, specify 'state' as 'Enabled' and 'isAzureMonitorTargetEnabled' as true  When using REST API to configure auditing, Diagnostic Settings with 'SQLSecurityAuditEvents' diagnostic logs category on the database should be also created Note that for server level audit you should use the 'master' database as {databaseName}  Diagnostic Settings URI format: PUT https://managementazurecom/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/MicrosoftSql/servers/{serverName}/databases/{databaseName}/providers/microsoftinsights/diagnosticSettings/{settingsName}?api-version=2017-05-01-preview  For more information, see [Diagnostic Settings REST API](https://gomicrosoftcom/fwlink/?linkid=2033207) or [Diagnostic Settings PowerShell](https://gomicrosoftcom/fwlink/?linkid=2033043)",
```
Comment (roneli) [inline resources/sql_databases.go:381]: shorten description

File: resources/sql_databases.go [hunk 6]
```diff
@@ -316,6 +316,97 @@ func SQLDatabases() *schema.Table {
...
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "kind",
+						Description: "Resource kind",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "state",
+						Description: "Specifies the state of the policy If state is Enabled, storageEndpoint or isAzureMonitorTargetEnabled are required Possible values include: 'BlobAuditingPolicyStateEnabled', 'BlobAuditingPolicyStateDisabled'",
```
Comment (roneli) [inline resources/sql_databases.go:339]: shorten description

File: resources/sql_databases.go [hunk 7]
```diff
@@ -316,6 +316,97 @@ func SQLDatabases() *schema.Table {
...
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("DatabaseBlobAuditingPolicyProperties.State"),
+					},
+					{
+						Name:        "storage_endpoint",
+						Description: "Specifies the blob storage endpoint (eg https://MyAccountblobcorewindowsnet) If state is Enabled, storageEndpoint or isAzureMonitorTargetEnabled is required",
```
Comment (roneli) [inline resources/sql_databases.go:345]: shorten description

**Review by roneli — CHANGES_REQUESTED** [review]
Minor comments, mostly to reduce long descriptions

File: resources/sql_databases.go (hunk 7 above)
Comment (irmatov) [reply resources/sql_databases.go:345]: Done

File: resources/sql_databases.go (hunk 1 above)
Comment (irmatov) [reply resources/sql_databases.go:363]: Done

File: resources/sql_databases.go (hunk 6 above)
Comment (irmatov) [reply resources/sql_databases.go:339]: Done

File: resources/sql_databases.go (hunk 5 above)
Comment (irmatov) [reply resources/sql_databases.go:381]: Done

File: resources/sql_servers.go (hunk 4 above)
Comment (irmatov) [reply resources/sql_servers.go:309]: Done

File: resources/sql_servers.go (hunk 3 above)
Comment (irmatov) [reply resources/sql_servers.go:321]: Done

File: resources/sql_servers.go (hunk 2 above)
Comment (irmatov) [reply resources/sql_servers.go:339]: Done

**Review by roneli — COMMENTED** [review]
New SDK supports INET/Mac/IP types we should change to the correct types if possible.

**Comment by irmatov** [comment]
@roneli I don't think this PR adds any fields of these types in new resources

**Review by roneli — APPROVED** [review]

**Pull request merged**